File: projects/09/clojure/src/find.clj
;; find.clj tests a function to search a list for a value.
;;
;; Output: four lists, and 
;;          the results of calling a listSearch() function on them.
;;
;; Usage: clojure -m find
;;
;; Begun by: Dr. Adams, CS 214 at Calvin College.
;; Completed by: Quentin Barnes
;; Date: May 6, 2019
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns find)

(load "listOps")                      ; load module

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;; Function -main tests function listSearch(). 
;; Output: the results of testing listSearch() on various lists.
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
(defn -main []
  (let
    [ 
      list1 '(99 88 77 66 55)        ; 99 is first value
      list2 (list 55 66 77 88 99)    ; 99 is last value
      list3 (list 55 77 99 88 66)    ; 99 is middle value
      list4 (list 55 77 45 88 66)    ; 99 doesnt exist
    ]

    (println)
    (println "list1: " list1)        ; print list1,
    (println "list2: " list2)        ;   list2,
    (println "list3: " list3)        ;   list3,
    (println "list4: " list4)        ;   list4,
    (println)                        ;   and their search
    (println "The position of 99 in list1 is: " (listSearch list1 99 0))
    (println "The position of 99 in list2 is: " (listSearch list2 99 0))
    (println "The position of 99 in list3 is: " (listSearch list3 99 0))
    (println "The position of 99 in list4 is: " (listSearch list4 99 0))
    (println)
  )
)

File: projects/09/ruby/find.rb
# fin.rb tests a Ruby linked list.
#
# Begun by: Dr. Adams for CS 214 at Calvin College.
# Completed by: Quentin Barnes
# Date: May 6, 2019

require './List'

list1 = List.new
list2 = List.new
list3 = List.new
list4 = List.new

list1.append(99).append(88).append(77).append(66).append(55)
list2.append(55).append(66).append(77).append(88).append(99)
list3.append(55).append(77).append(99).append(88).append(66)
list4.append(55).append(77).append(45).append(88).append(66)

list1.print 
puts
puts "Position of 99 in list1: #{list1.search(99)}"

list2.print 
puts
puts "Position of 99 in list2: #{list2.search(99)}"

list3.print 
puts
puts "Position of 99 in list3: #{list3.search(99)}"

list4.print 
puts
puts "Position of 99 in list4: #{list4.search(99)}"

